### PR TITLE
ext_authz: Log ext_authz gRPC messages at trace level

### PR DESCRIPTION
**Pull request by ramaraochavali**
Commit Message: log external auth request/response proto messages
Additional Description: Logs ext auth request/response proto messages for debugging at trace level.

Risk Level: Low
Testing: N/A
Docs Changes: N/A
Release Notes: N/A
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Deprecated:]

### CODE REVIEW

File: source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc [hunk 1]
```diff
@@ -41,12 +41,14 @@ void GrpcClientImpl::check(RequestCallbacks& callbacks,
   ASSERT(callbacks_ == nullptr);
   callbacks_ = &callbacks;
 
+  ENVOY_LOG(trace, "Sending CheckRequest : {}", request.DebugString());
```
Comment (dio) [inline source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc:44]: remove superfluous space before `:`?

```
ENVOY_LOG(trace, "Sending CheckRequest: {}", request.DebugString());
```

File: source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc [hunk 2]
```diff
@@ -41,12 +41,14 @@ void GrpcClientImpl::check(RequestCallbacks& callbacks,
...
   request_ = async_client_->send(service_method_, request, *this, parent_span,
                                  Http::AsyncClient::RequestOptions().setTimeout(timeout_));
 }
 
 void GrpcClientImpl::onSuccess(std::unique_ptr<envoy::service::auth::v3::CheckResponse>&& response,
                                Tracing::Span& span) {
+  ENVOY_LOG(trace, "Received CheckResponse : {}", response->DebugString());
```
Comment (dio) [inline source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc:51]: remove superfluous space before `:`?

File: source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc [hunk 3]
```diff
@@ -73,6 +75,8 @@ void GrpcClientImpl::onSuccess(std::unique_ptr<envoy::service::auth::v3::CheckRe
 
 void GrpcClientImpl::onFailure(Grpc::Status::GrpcStatus status, const std::string&,
                                Tracing::Span&) {
+  ENVOY_LOG(trace, "CheckRequest call failed with status : {}",
```
Comment (dio) [inline source/extensions/filters/common/ext_authz/ext_authz_grpc_impl.cc:78]: remove superfluous space before `:`?

**Comment by ramaraochavali** [comment]
@dio Adjusted. PTAL

**Pull request merged**